include/dsn/c/api_layer1.h

```diff
@@ -578,7 +578,7 @@ typedef union dsn_global_partition_id
 
 inline uint64_t dsn_gpid_to_hash(dsn_gpid gpid)
 {
-    return (((uint64_t)gpid.u.app_id) << 32) + gpid.u.partition_index;
+    return static_cast<uint64_t>(gpid.u.app_id ^ gpid.u.partition_index);
```

Comment:
best to use the original hash as: (1) now the hash is for both partition and thread localization; (2) the former hash has better distribution in terms of load balance for the partitions.